### ISSUE TITLE
qtpylib.broker: Not submitting AAPL order, orders pending (1.5.76)

### ISSUE THREAD

**Issue by mariope** [ranaroussi/qtpylib#71]
I'm getting "Not submitting AAPL order, orders pending" with 1.5.76 version and all dependences updated with DumbAlgo strategy.

```
2017-10-25 18:16:01,077 [WARNING] qtpylib.broker: Not submitting AAPL order, orders pending: {'AAPL': {'orderId': 1, 'quantity': 100, 'expires': datetime.datetime(2017, 10, 25, 18, 16, 1, 216463)}}
2017-10-25 18:18:01,013 [WARNING] qtpylib.broker: Not submitting AAPL order, orders pending: {'AAPL': {'orderId': 1, 'quantity': 100, 'expires': datetime.datetime(2017, 10, 25, 18, 18, 1, 855148)}}
2017-10-25 18:20:01,555 [WARNING] qtpylib.broker: Not submitting AAPL order, orders pending: {'AAPL': {'orderId': 1, 'quantity': 100, 'expires': datetime.datetime(2017, 10, 25, 18, 20, 2, 312194)}}
2017-10-25 18:23:01,185 [WARNING] qtpylib.broker: Not submitting AAPL order, orders pending: {'AAPL': {'orderId': 1, 'quantity': 100, 'expires': datetime.datetime(2017, 10, 25, 18, 23, 1, 825973)}}
```

**Comment by acennami** [comment]
Having same issue on a new installation today.  Using ezibpy 1.12.56.  Using a paper trading account with TWS 968.2e (local).

Using the DumbAlgo strategy, and I can see the BUY trigger for several minutes and then get the error messages, with no successful orders entered.


(qtpylib) acennami@wqdev01:~/projects/trading/qtpylib$ python strategy.py --log ~/projects/trading/qtpylib/logs/
[BAR] AMD datetime: 2017-10-26 09:14:01.202802
[FLAT] AMD - Execute Buy
[BAR] AMD datetime: 2017-10-26 09:15:07.348774
[FLAT] AMD - Execute Buy
[BAR] AMD datetime: 2017-10-26 09:16:19.267111
[FLAT] AMD - Execute Buy
[BAR] AMD datetime: 2017-10-26 09:17:38.880639
[FLAT] AMD - Execute Buy
[BAR] AMD datetime: 2017-10-26 09:18:16.842752
[FLAT] AMD - Execute Buy
2017-10-26 09:18:16,843 [WARNING] qtpylib.broker: Not submitting AMD order, orders pending: {'AMD': {'orderId': 1, 'quantity': 100, 'expires': datetime.datetime(2017, 10, 26, 9, 18, 38, 882597)}}
[BAR] AMD datetime: 2017-10-26 09:19:23.451164
[FLAT] AMD - Execute Buy
[BAR] AMD datetime: 2017-10-26 09:20:06.167367
[FLAT] AMD - Execute Buy

**Comment by mariope** [comment]
same here on new installation:
TWS 968.2e
qtpy 1.5.76
ezibpy 1.12.56
macOS HS 10.13

all qtpylib requirements are already up-to-date.

There are identical issues with older version that were closed with an update recommendation and no further comments: 

- [#63](https://github.com/ranaroussi/qtpylib/issues/63)
- [#66](https://github.com/ranaroussi/qtpylib/issues/66)

**Comment by acennami** [comment]
Well, good news is I figured out what the issue was.  How to fix it correctly is another issue altogether.

Whether its passed through CLI, script or manually modifying the default arguments, Broker() is not inheriting a modified 'ibserver' argument from Algo().

Initially I was not modifying the argument myself, assuming that when it was setup via Blotter() it would be inherited in the other classes.  But once I realized that nothing was hitting the API, I updated my strategy.py to include the modified location for TWS/IGBW.  

When I modified the 'ibport' argument in strategy.py I saw that was being passed correctly, but no matter how or where i modified 'ibserver' it would not be inherited by Broker(), so Broker wasn't able to establish the API connection.

I'm a novice with Python and programming in general, so that's about as much as I could provide, and couldn't really say where or how to fix it - other than editing broker.py manually in the library with the correct IP and port fixed the issue for me, and I'm now up and running paper trading.

This framework seems far more developed and better documented than most of the others I've toyed with, so I'm very excited to begin actually playing around with it.  

Thanks!
